runtime/src/accounts_index.rs

```diff
@@ -173,14 +173,154 @@ impl<T: 'static + Clone + IsCached> WriteAccountMapEntry<T> {
     }
 }
 
+use bv::BitVec;
+
 #[derive(Debug, Default)]
+pub struct RollingBitField {
+    max_width: u64,
+    min: u64,
+    max: u64, // exclusive
+    bits: BitVec,
+    count: usize,
+}
+// functionally similar to a hashset
+// Relies on there being a sliding window of key values. The key values continue to increase.
+// Old key values are removed from the lesser values and do not accumulate.
+impl RollingBitField {
+    pub fn new(max_width: u64) -> Self {
+        assert!(max_width > 0);
+        assert!(max_width.is_power_of_two()); // power of 2 to make dividing a shift
+        let bits = BitVec::new_fill(false, max_width);
+        Self {
+            max_width,
+            bits,
+            count: 0,
+            min: 0,
+            max: 0,
+        }
+    }
+
+    fn get_address(&self, key: &u64) -> u64 {
+        Self::calc_address(self.max_width, key)
+    }
+
+    // find the array index
+    fn calc_address(max_width: u64, key: &u64) -> u64 {
+        key % max_width
```

Comment:
Function seems a bit superfluous

Reply:
yep, was left over from when it was more complicated and the static version was called prior to construction